src/plugins/hiddenColumns/hiddenColumns.js

```diff
@@ -269,49 +272,28 @@ class HiddenColumns extends BasePlugin {
     arrayEach(this.hot.getCellsMeta(), (meta) => {
       if (meta) {
         meta.skipColumnOnPaste = false;
-
-        if (meta.baseRenderer !== null) {
-          meta.renderer = meta.baseRenderer;
-          meta.baseRenderer = null;
-        }
       }
     });
   }
 
   /**
-   * Sets width hidden columns on 0.
+   * Adds the additional column width for the hidden column indicators.
    *
    * @private
    * @param {number} width Column width.
-   * @param {number} column Column index.
+   * @param {number} column Visual column index.
    * @returns {number}
    */
-  onBeforeStretchingColumnWidth(width, column) {
-    let stretchedWidth = width;
-
+  onModifyColWidth(width, column) {
+    // Hook is triggered internally only for the visible columns. Conditional will be handled for the API
+    // calls of the `getColWidth` function on not visible indexes.
     if (this.isHidden(column)) {
-      stretchedWidth = 0;
+      return 0;
     }
 
-    return stretchedWidth;
-  }
+    if (this.settings.indicators && (this.isHidden(column + 1) || this.isHidden(column - 1))) {
 
-  /**
-   * Adds the additional column width for the hidden column indicators.
-   *
-   * @private
-   * @param {number} width Current column width.
-   * @param {number} col Visual column index.
-   * @returns {number}
-   */
-  onModifyColWidth(width, col) {
-    if (this.isHidden(col)) {
-      return 0.1;
-
-    } else if (this.settings.indicators && (this.isHidden(col + 1) ||
-               this.isHidden(col - 1))) {
-
-      // add additional space for hidden column indicator
+      // Add additional space for hidden column indicator.
       return width + (this.hot.hasColHeaders() ? 15 : 0);
```

Comment:
Out of the scope but since we did a refactor, why not check `hasColHeaders` first? If it's false we don't have to validate all other conditions

Reply:
Done: https://github.com/handsontable/handsontable/pull/6547/commits/f25f7a194105994a725f43d95ec60f85ec113d2d#diff-8bdb2c73d5902c39670238216b4027be.